osg-pilot-container/osgpilot_meter

```diff
@@ -440,6 +459,30 @@ def detect_new_jobs(sqldb, current_ts):
     DebugPrint(INFO, "- deleted %s rows from inbox" % rc2)
 
 
+def detect_and_update_partial_jobs(sqldb):
+    DebugPrint(INFO, "Reporting and breaking up partial jobs")
+
+    insert_sql = """
+        insert into outbox (%s)
+        select *
+          from cutoff
+    """ % ", ".join(SEND_ATTRS)
+
+    # reset aggregates
+    update_sql = """
+        update jobs
+           set DaemonStartTime = last_updated,
+               total_cpu = 0,
+               MemoryUsage = 0
+         where last_updated - DaemonStartTime >= {partial_record_cutoff};
+    """.format(partial_record_cutoff=PARTIAL_RECORD_CUTOFF)
+
+    rc1 = sqldb.execute(insert_sql, (dead_cutoff,)).rowcount
```

Comment:
I know that we most likely are only running a single instance at a time but can you humor the computer scientist in me and do this statement and the next one in a single transaction?

Reply:
This is done in a single transaction; the commit comes later ([here](https://github.com/opensciencegrid/gratia-probe/pull/83/files#diff-f5aee1b71bdecdb74c1a431acc1770cc2171e74573ef0ae294f57732fce5a10bR502)).